### DISCUSSION TITLE
Could we have a feature to delete unwanted pin_group and ports?

### DISCUSSION THREAD

**Discussion by raindigoa** [ansys/pyaedt#2443] [Ideas]
Hi AE,

Thank you for your assistance last time.

This time I bring a new idea which is extended by the last one.

Now we can have the user-defined pin_groups and corresponding ports in pyaedt, it's great.

But it seems there are no functions supporting the "delete" action in pyaedt.

For example, if I create a pin group "A1" with the pins [a1,a2,a3,a4]

and I suddenly notice that I missed another pins a5-a10, but they all belong to "A1" group

Now I cannot remove the wrong pin group by pyaedt command.

Likely, if I set ports with the wrong pin groups unconsciously, I have no way to delete the wrong ports by pyaedt command.

Would you please add this function for both pin group and port application?

thank you

**Comment by svandenb-dev** [comment]
Hi,
thank you for pointing this issue. Are you working with edb module ? I can add this feature into it.
I will convert this thread into Issue on GitHub.
Thanks

**Reply by raindigoa** [reply]
Hi,

Yes, I am working with edb module

thank you

**Comment by hui-zhou-a** [comment]
@raindigoa @svandenb-dev I am not sure if it is a good idea to delete/modify pin groups. If something is wrong, it is every easy to modify the script to create a new project from the original edb. Delete/modifying pin groups is not as simple as it sounds. There could be a source terminal created on the pin group, for instance. All these related objects must be found and removed.

**Reply by raindigoa** [reply]
Hi @ring630 
I think it would be okay for a terminal lost its pin group just like what we would do unintentionally in SIwave sometimes. You can always remove a group while keep corresponding port, and vice versa.
Just as what I reported in this post https://github.com/pyansys/pyaedt/issues/2442
Now the thing is that we may probably set ports with the wrong pin groups(even pyaedt would make mistakes like this), but we have no way to remove it under the scripting scenario, it's quite frustrating.
I think it is not good for users to find out the port setting issues after they save, close and import the edb file into the ansys tool.
My recommendation is that you can add the attribution to the port objects to let the users know what the positive/negative terminal is, so that they could choose not to save the edb file as soon as they find the error. And add more functions in the future.

**Reply by hui-zhou-a** [reply]
@raindigoa I have read through your post 2442. It looks like a bug, which I saw @svandenb-dev is working on it already. Ideally, all ports should be set correctly accoring to the script, and there is no need to check again visually. Anyway,  @svandenb-dev is working on pin group removal already. Thanks for your comments.

**Reply by raindigoa** [reply]
Hi @ring630 
Thank you for the update, @svandenb-dev appreciate your works!
